aws/resource_aws_ssm_parameter.go

```diff
@@ -47,45 +52,88 @@ func resourceAwsSsmParameter() *schema.Resource {
 			"key_id": {
 				Type:     schema.TypeString,
 				Optional: true,
-				ForceNew: true,
 			},
 			"overwrite": {
 				Type:     schema.TypeBool,
 				Optional: true,
-				Default:  false,
 			},
+			"allowed_pattern": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"tags": tagsSchema(),
 		},
 	}
 }
 
+func resourceAwsSmmParameterExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	ssmconn := meta.(*AWSClient).ssmconn
+
+	resp, err := ssmconn.GetParameters(&ssm.GetParametersInput{
+		Names:          []*string{aws.String(d.Id())},
+		WithDecryption: aws.Bool(true),
+	})
+
+	if err != nil {
+		return false, err
+	}
+	return len(resp.InvalidParameters) == 0, nil
+}
+
 func resourceAwsSsmParameterRead(d *schema.ResourceData, meta interface{}) error {
 	ssmconn := meta.(*AWSClient).ssmconn
 
 	log.Printf("[DEBUG] Reading SSM Parameter: %s", d.Id())
 
-	paramInput := &ssm.GetParametersInput{
-		Names: []*string{
-			aws.String(d.Id()),
-		},
+	resp, err := ssmconn.GetParameters(&ssm.GetParametersInput{
+		Names:          []*string{aws.String(d.Id())},
 		WithDecryption: aws.Bool(true),
+	})
+	if err != nil {
+		return errwrap.Wrapf("[ERROR] Error getting SSM parameter: {{err}}", err)
```

Comment:
Minor nitpick: We don't need to use `errwrap.Wrapf` normally if we are just returning an error message, can simply be `return fmt.Errorf("error getting SSM parameter: %s", err)`